File: DependencyInjection/DependencyInjection/_1_ControlFreak/Outbound/XmlOutboundMessage.cs
using DependencyInjectionBefore._1_ControlFreak.Interfaces;

namespace DependencyInjectionBefore._1_ControlFreak.Outbound
{
  class XmlOutboundMessage : DataDestination
  {
    private readonly XmlMarshalling _xmlMarshalling;
    private string _content = string.Empty;

    public XmlOutboundMessage()
    {
      _xmlMarshalling = new XmlMarshalling();
    }

    public void SendVia(TcpSocket outputSocket)
    {
      var marshalledContent = _xmlMarshalling.Of(_content);
      outputSocket.Open();
      outputSocket.Send(marshalledContent);
      outputSocket.Close();
    }

    public void Add(string s)
    {
      _content += s;
    }
  }
}

File: Java/dependency-injection/_02_control-freak/src/main/java/com/github/grzesiek_galezowski/DependencyInjection/_1_ControlFreak/Outbound/XmlOutboundMessage.java
package com.github.grzesiek_galezowski.DependencyInjection._1_ControlFreak.Outbound;

import com.github.grzesiek_galezowski.DependencyInjection._1_ControlFreak.Interfaces.DataDestination;

public class XmlOutboundMessage implements DataDestination {
  private final XmlMarshalling _xmlMarshalling;
  private String content = "";

  public XmlOutboundMessage() {
    _xmlMarshalling = new XmlMarshalling();
  }

  public void sendVia(TcpSocket outputSocket) {
    String marshalledContent = _xmlMarshalling.of(content);
    outputSocket.open();
    outputSocket.send(marshalledContent);
    outputSocket.close();
  }

  public void add(String s) {
    content += s;
  }
}
